### PR TITLE
Convert noteblocks for learn/javascript folder

### PR DESCRIPTION
**Pull request by queengooborg**
This PR converts the noteblocks for the 'learn/javascript' folder to GFM syntax, using a [conversion script](https://github.com/queengooborg/mdn-toolkit/blob/main/upgrade-noteblock.js).

### CODE REVIEW

**Comment by Josh-Cena** [comment]
@queengooborg question: is there a rendering difference between `> [!NOTE]\n> Content` and `> [!NOTE]\n>\n> Content`?

**Comment by queengooborg** [comment]
Nope, no rendering difference!  The empty line in between is usually to mitigate a Prettier bug when the line starts with a special character (like [ or `).

File: files/en-us/learn/javascript/first_steps/useful_string_methods/index.md [hunk 1]
```diff
@@ -163,7 +163,9 @@ browserType.slice(2); // "zilla"
 
 This returns `"zilla"` — this is because the character position of 2 is the letter `"z"`, and because you didn't include a second parameter, the substring that was returned was all of the remaining characters in the string.
 
-> **Note:** `slice()` has other options too; study the {{jsxref("String.prototype.slice()", "slice()")}} page to see what else you can find out.
+> [!NOTE]
+>
+> `slice()` has other options too; study the {{jsxref("String.prototype.slice()", "slice()")}} page to see what else you can find out.
```
Comment (Josh-Cena) [inline files/en-us/learn/javascript/first_steps/useful_string_methods/index.md:168]:
```suggestion
> **Note:** `slice()` has other options too; study the {{jsxref("String.prototype.slice()", "slice()")}} page to see what else you can find out.
```

**Review by Josh-Cena — CHANGES_REQUESTED** [review]
https://github.com/mdn/content/pull/35067

**Review by Josh-Cena — APPROVED** [review]

**Pull request merged**